### PR TITLE
feat(releases) Collect timing information on release commit creation

### PR DESCRIPTION
**Pull request by markstory**
We have a few internal Issues that point to either our locks not being long enough or something worse. Before moving forward with #11014 it would be good to confirm or rule out my hunch about locks expiring.

### CODE REVIEW

**Review by mattrobenolt — COMMENTED** [review]
I won't block on this, but just a thought: what if we added this into `TimedRetryPolicy` natively?

And trigger sending a timing if you do `TimedRetryPolicy(10, metric_name='release.set_commits.duration', metric_tags={})` or whatever, and automatically report when `metric_name is not None`.

That way we can more easily track lock timings just by giving it a name.

**Comment by markstory** [comment]
@mattrobenolt Tracking how many times we spin on a lock.acquire would be great. 

My current theory on what is happening is:

1. Process 1 gets lock, starts syncing commits.
2. Process 2 starts a few seconds later and starts spinning on the lock
3. Process 1's lock expires, but the work is not complete, allowing Process 2 to get the lock.
4. Process 1 finishes its work, and then Process 2 hits integrity errors as it attempts to repeat the same work.

My thinking is that with this new metric I can determine if the workload takes longer than the lock duration of 10 seconds.

File: src/sentry/models/release.py [hunk 1]
```diff
@@ -452,6 +454,7 @@ def set_commits(self, commit_list):
                     ],
                     last_commit_id=latest_commit.id if latest_commit else None,
                 )
+                metrics.timing('release.set_commits.duration', time() - start)
```
Comment (wedamija) [inline src/sentry/models/release.py]: Nbd, but we could use the `metrics.timer` context manager instead of calculating manually

**Review by wedamija — APPROVED** [review]

File: src/sentry/models/release.py (hunk 1 above)
Comment (markstory) [reply src/sentry/models/release.py]: I didn't want to diff 300 lines for a 1 line change :blush:

Comment (wedamija) [reply src/sentry/models/release.py]: Fair :P

**Pull request merged**